src/Bridge/Doctrine/Persister/ObjectManagerPersister.php

```diff
@@ -42,9 +44,9 @@
      */
     private $metadata = [];
 
-    public function __construct(ObjectManager $manager)
+    public function __construct(ManagerRegistry $managerRegistry)
```

Comment:
Maybe a more BC friendly way would be to inject directly the managers:

```php
function __construct(ObjectManager ...$managers) {
  $this->managers = $managers;
}
```

Since the service is lazy it shouldn't cause any issue. This may require an extra dependency with the expression language though

Reply:
Having the `ManagerRegistry` allow to call `$this->managerRegistry->getManagerForClass($class);` directly, wich remove a lot of complexity. See below

Reply:
Yeah @jdeniau is right, having the `ManagerRegistry` is nice to remove some complexity.

What should we do ? `ObjectManager ...$managers` with some complexity but no BC or `ManagerRegistry $managerRegistry` with reduced complexity but BC. Up to you @theofidry .

Reply:
can the registry be instantiated here since we cannot inject it?

Reply:
instantiated ? You mean make a `new ManagerRegistry()` inside the constructor ? it seems hard, see the service declaration : https://github.com/doctrine/DoctrineBundle/blob/master/Resources/config/dbal.xml#L59L65

Reply:
Looking at the implementation: https://github.com/doctrine/persistence/blob/master/lib/Doctrine/Common/Persistence/AbstractManagerRegistry.php#L156

I don't think iterating over the managers is a bad idea in the end. Maybe you can create a custom `ManagerRegistry` class to be able to cache the results a bit like we are doing for the metadata, which you then can instantiate more easily

Reply:
Thinking about it again, what about have a dedicated class implementing `ObjectManager`?

```php
class ObjectsManager implements ObjectManager {
  function (ManagerRegistry $registry) {
    $this->registry = $registry;
  }

  function getClassMetadata(string $class): Metadata {
    $manager = $this->managerRegistry->getManagerForClass($class)
    $manager->getClassMetadata($class);
  }
}
```

So that manager can take the registry just fine and we can inject this new manager without any BC breaks

Reply:
I think `ObjectManager[]` would be the best approach since it allows to push managers from both ORM and ODM, you can treat them the same regardless where they came from.

Reply:
Hi. I don't know if my comment would help but injecting the registry and getting a fresh manager each time you want to persist will also allow insert errors. Usually as soon as the database throws an error, the manager closes. The only way to re-open the connection is to get a fresh manager from the registry. Keeping in mind this I would suggest to inject the registry instead of all the managers.